### ISSUE TITLE
Display multiple circuits and voltages

### ISSUE THREAD

**Issue by Gazer75** [openinframap/styles#8]
Would be nice if it would somehow show that a line has more than one circuit.

Also making use of **voltage:primary** and **voltage:secondary** would be possible display both voltages. Maybe by making a dashed alternating colour?

The two keys above were only mentioned on the wiki for things like transformers, but I see no reason not to use them on dual circuit lines as well.

**Comment by Gazer75** [comment]
Unless someone changes the tags then here is a nice way to test on:
https://www.openstreetmap.org/way/549830974

**Comment by gustavecha** [comment]
Hi

Voltage:primary, voltage:secondary shouldn't be used on power lines
This tagging is intended for transformers only
primary, secondary, tertiary explicitly refer to transformer interfaces.

voltage=132000;66000 is the only appropriate solution for your exemple

**Comment by Gazer75** [comment]
Really don't like the idea of separating values with semicolon. This was also the way to describe transformers before.

**Comment by gustavecha** [comment]
I agree the semicolon solution is not the best and this is precisely why voltage:primary, secondary... were intended on transformers only.
There are no interfaces on power lines to deal with.

**Comment by Gazer75** [comment]
Well it doesn't have to be an interface. Guess to avoid confusion one could do **line:voltage:primary** or something similar

**Comment by gustavecha** [comment]
I understand this, nevertheless it's not so simple.
Primary, Secondary, ... are terms precisely defined by IEC and they are related to transformers, not to power lines
http://www.electropedia.org/iev/iev.nsf/display?openform&ievref=421-03-06

This tagging has been reviewed few weeks ago
https://wiki.openstreetmap.org/wiki/Proposed_features/Transformer_extension_proposal

Finally I don't see point to have **sorted** (secondary is after primary) list of voltages for lines while a given line is built for a given voltage and then, some circuits can be operated at lower voltage (that's why 66000 v can be found on a 132000 v line)
The best solution would be to refine voltage=* key for lines and give only the builtin voltage (here 132 kV) and involve the way in separate relations for 132 kV and 66kV circuit).
Example here :
* 400000 power line : https://www.openstreetmap.org/way/426020692 (voltage=400000;225000 but it may be voltage=400000 since the both circuits are built for 400 kV)
* 400000 power circuit : https://www.openstreetmap.org/relation/6333475
* 225000 power circuit : https://www.openstreetmap.org/relation/6879070
 
But there's no point to sort them between primary and secondary

**Comment by russss** [comment]
I echo @gustavecha's comments that primary/secondary could be confusing. 

I think keeping the voltage in the semicolon-delimited voltage= tag is still the most "OSM-like" approach (it's analogous to the way it works for substations, for example). I'm not entirely sure how I would handle multiple voltages like this but I'll have a think about it.

There is a related issue here, where a single set of pylons carries two different circuits of the same voltage - this is quite common in the UK transmission network. Each circuit can have a different source and/or destination substation, and hence a different name and reference.

I think relations are a good way to store this data, but rendering it is a different matter...

**Comment by gustavecha** [comment]
An exemple of such rendering, considering only ways and not relation can be seen here: 
https://github.com/gustavecha/openinframap-styles/blob/f3af96b1ce9d46d688467abe1fd52a73d780af61/josm/power_fr.mss

Just filter on way [power=line][cables=6] or way [power=line][circuits=2] to display extra casing, dashes or whatever
Colors remain as an issue since voltage=* values have to be parsed for each country or network grid specifically as you will see in the mss file

**Comment by Gazer75** [comment]
OK I'll just stick with semicolon.

I have to admit I do not understand the coding behind it all very well.
So if some of this is already in the file then please ignore me :)

Maybe the labels could also help indicate that there are multiple circuits?
Say if its two of the same then the voltage part would display 2x voltage
If it was two different ones it could maybe say voltage / voltage

Why do you have to do this for each country?

**Comment by pyrog** [comment]
It work nice for lines with different voltages :) but not (yet) for lines with only one voltage:

- [way/549830974
![Screenshot_2019-06-06 Open Infrastructure Map(3)](https://user-images.githubusercontent.com/2158081/59061859-c3a69880-88a4-11e9-8aeb-64b2c68b0b45.png)](https://openinframap.org/#17.9/60.605686/9.076568)

- [way/254671324
![Screenshot_2019-06-06 Open Infrastructure Map(4)](https://user-images.githubusercontent.com/2158081/59062175-84c51280-88a5-11e9-8418-ead15c8c43cf.png)
](https://openinframap.org/#16.72/46.348227/5.632934)

**Comment by russss** [comment]
Yes - I don't currently import the `circuits=` tag so that's waiting for a reimport.

**Comment by pyrog** [comment]
Ok. See also #49

**Comment by russss** [comment]
Oh, this got fixed ages ago. If there are any further issues, feel free to open a new ticket.
